### PR TITLE
Fix db name too long at subCacheRefresh issue#2848

### PR DESCRIPTION
**Pull request by ArqamFarooqui110719**
Fix to add check for db name too long at subCacheRefresh issue #2848

### CODE REVIEW

**Comment by ArqamFarooqui110719** [comment]
Hi @fgalan Sir, There is one FT is failing related to issue #2811 i.e. `2811_long_service_names_in_subscriptions/long_service_names_in_subscriptions.test`:
TestCase#4:
```
04. grep in log file to make sure there are NO ERRORS
```
Current behavior: (No error in log file)
```
 04. grep in log file to make sure there are NO ERRORS
 =====================================================
```
 
Expected Behavior after fix:
```
 04. grep in log file to make sure there are NO ERRORS
 =====================================================
+time=2023-03-15T11:45:24.510Z | lvl=ERROR | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=MongoGlobal.cpp[263]:getOrionDatabases | msg=Runtime Error (database name size should be smaller than 50 characters: ftest-t012345678901234567890123456789012345678901234)
```
Should I need to remove the TestCase#4 or I need to update the TestCase#4? Please let me know your opinion?

File: src/lib/mongoBackend/MongoGlobal.cpp [hunk 1]
```diff
@@ -251,9 +251,18 @@ bool getOrionDatabases(std::vector<std::string>* dbsP)
...
+      if (strlen(dbName.c_str()) <= 50)
+      {
+        LM_T(LmtMongo, ("Orion database found: %s", dbName.c_str()));
+        dbsP->push_back(dbName);
+        LM_T(LmtBug, ("Pushed back db name '%s'", dbName.c_str()));
+      }
+      else
+      {
+        LM_E(("Runtime Error (database name size should be smaller than 50 characters: %s)", dbName.c_str()));
+        return false;
```
Comment (fgalan) [inline src/lib/mongoBackend/MongoGlobal.cpp:264]:
```suggestion
```

We don't want a "bad" database ruins all the cache refresh. Just skipping it (with the proper LM_E error) is fine, and the for loop will continue.

File: src/lib/mongoBackend/MongoGlobal.cpp [hunk 2]
```diff
@@ -251,9 +251,18 @@ bool getOrionDatabases(std::vector<std::string>* dbsP)
 
     if (strncmp(prefix.c_str(), dbName.c_str(), strlen(prefix.c_str())) == 0)
     {
-      LM_T(LmtMongo, ("Orion database found: %s", dbName.c_str()));
-      dbsP->push_back(dbName);
-      LM_T(LmtBug, ("Pushed back db name '%s'", dbName.c_str()));
+      // Check for size of database name, db names size should be <= '50' bytes.
+      if (strlen(dbName.c_str()) <= 50)
```
[the remark this answers is not in the document]
Comment (ArqamFarooqui110719) [reply src/lib/mongoBackend/MongoGlobal.cpp]: Fixed in [8926692](https://github.com/telefonicaid/fiware-orion/pull/4297/commits/8926692de1eb33c90acbefc49dcd180488cba335)

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -3,4 +3,5 @@
 - Fix: Ngsiv2-AttrsFormat header set to "normalized" in NGSI patching custom notification
 - Fix: memory access potential problem in GET /log/trace request when tracelevel 255 is enabled
 - Fix: Orion should return error and refuse to start when port is negative (#3875)
-- Fix: lineMaxSize and infoPayloadMaxSize in the log admin REST API (#3707)
\ No newline at end of file
+- Fix: lineMaxSize and infoPayloadMaxSize in the log admin REST API (#3707)
+- Fix: Check for service name too long at subCacheRefresh (#2848)
```
[the remark this answers is not in the document]
Comment (ArqamFarooqui110719) [reply CHANGES_NEXT_RELEASE]: Fixed in [8926692](https://github.com/telefonicaid/fiware-orion/pull/4297/commits/8926692de1eb33c90acbefc49dcd180488cba335)

File: src/lib/mongoBackend/MongoGlobal.cpp [hunk 4]
```diff
@@ -251,9 +251,18 @@ bool getOrionDatabases(std::vector<std::string>* dbsP)
...
+      {
+        LM_T(LmtMongo, ("Orion database found: %s", dbName.c_str()));
+        dbsP->push_back(dbName);
+        LM_T(LmtBug, ("Pushed back db name '%s'", dbName.c_str()));
+      }
+      else
+      {
+        LM_E(("Runtime Error (database name size should be smaller than 50 characters: %s)", dbName.c_str()));
```
[the remark this answers is not in the document]
Comment (ArqamFarooqui110719) [reply src/lib/mongoBackend/MongoGlobal.cpp]: Fixed in [8926692](https://github.com/telefonicaid/fiware-orion/pull/4297/commits/8926692de1eb33c90acbefc49dcd180488cba335)

File: src/lib/mongoBackend/MongoGlobal.cpp (hunk 1 above)
Comment (ArqamFarooqui110719) [reply src/lib/mongoBackend/MongoGlobal.cpp]: Fixed in [8926692](https://github.com/telefonicaid/fiware-orion/pull/4297/commits/8926692de1eb33c90acbefc49dcd180488cba335)

**Comment by ArqamFarooqui110719** [comment]
> > Should I need to remove the TestCase#4 or I need to update the TestCase#4? Please let me know your opinion?
> 
> I think that after my suggestion of using DB_AND_SERVICE_NAME_MAX_LEN the fail in test will disappear.

Thanks for your suggestion:)
I have updated the PR, now all FTs are passing.
If the PR is OK, Kindly merge it.

**Pull request merged**